types/datum.go

```diff
@@ -1998,6 +2002,11 @@ func (d *Datum) ToMysqlJSON() (j json.BinaryJSON, err error) {
 	return
 }
 
+// MemUsage gets the memory usage of datum.
+func (d *Datum) MemUsage() (sum int64) {
+	return EmptyDatumSize + int64(cap(d.b)) + int64(len(d.collation)) + int64(unsafe.Sizeof(d.x)-16)
```

Comment:
`unsafe.Sizeof(d.x)` is determined in compile-time so it is always 16. Maybe we can remove `int64(unsafe.Sizeof(d.x)-16)`.

Reply:
Yes, remove d.x and add comments